src/trace/log.zig

```diff
@@ -1,366 +1,464 @@
-                return logger.field(name, value);
-            },
-            .noop => {
-                return .noop;
-            },
-        }
-    }
-
-    pub fn infof(self: Self, comptime fmt: []const u8, args: anytype) void {
-        switch (self) {
-            inline .standard, .test_logger => |logger| {
-                logger.infof(fmt, args);
-            },
-            .noop => {},
-        }
-    }
-
-    pub fn debugf(self: Self, comptime fmt: []const u8, args: anytype) void {
-        switch (self) {
-            inline .standard, .test_logger => |logger| {
-                logger.debugf(fmt, args);
-            },
-            .noop => {},
+/// A ScopedLogger could either be:
+/// - A StandardErrLogger
+/// - A TestingLogger
+pub fn ScopedLogger(comptime scope: ?[]const u8) type {
+    return union(LogKind) {
+        const Self = @This();
+        standard: *ChannelPrintLogger,
+        test_logger: *DirectPrintLogger,
+        noop: void,
+
+        /// Can be used in tests to minimize the amount of logging during tests.
+        pub const TEST_DEFAULT_LEVEL: Level = .warn;
+
+        pub fn unscoped(self: Self) Logger {
+            return switch (self) {
+                .standard => |logger| .{ .standard = logger },
+                .test_logger => |logger| .{ .test_logger = logger },
+                .noop => .noop,
+            };
         }
-    }
-
-    pub fn warnf(self: Self, comptime fmt: []const u8, args: anytype) void {
-        switch (self) {
-            inline .standard, .test_logger => |logger| {
-                logger.warnf(fmt, args);
-            },
-            .noop => {},
+        pub fn withScope(self: Self, comptime new_scope: []const u8) ScopedLogger(new_scope) {
+            return switch (self) {
+                .standard => |logger| .{ .standard = logger },
+                .test_logger => |logger| .{ .test_logger = logger },
+                .noop => .noop,
+            };
         }
-    }
 
-    pub fn errf(self: Self, comptime fmt: []const u8, args: anytype) void {
-        switch (self) {
-            inline .standard, .test_logger => |logger| {
-                logger.errf(fmt, args);
-            },
-            .noop => {},
+        pub fn deinit(self: *const Self) void {
+            switch (self.*) {
+                .standard => |*logger| {
+                    logger.*.deinit();
+                },
+                .test_logger => |*logger| {
+                    logger.*.deinit();
+                },
+                .noop => {},
+            }
         }
-    }
 
-    pub fn logf(self: Self, level: Level, comptime fmt: []const u8, args: anytype) void {
-        switch (self) {
-            inline .standard, .test_logger => |logger| {
-                logger.logf(level, fmt, args);
-            },
-            .noop => {},
+        pub fn err(self: *const Self) Entry {
+            return switch (self.*) {
+                .noop => Entry{ .noop = {} },
```

Comment:
Did we agree to just using `.noop` as value instead of `{}` (void) value?

Reply:
I think we agreed to use the `.noop` version. Updated in [15dcfda](https://github.com/Syndica/sig/pull/277/commits/15dcfda2ed37938c60e3643142591ebbef8ede7b)